internal/delete_handler.go

```diff
@@ -486,28 +521,28 @@ func getBeforeChoiceFunc(name string, modifier int) func(object BackupObject) bo
 }
 
 func getRetainChoiceFunc(retentionCount, modifier int) func(object BackupObject) bool {
-	count := 0
+	uniqueNames := map[string]bool{}
```

Comment:
In multi-storage world, backup name doesn't reference to a single backup anymore as multiple backups with the same name may exist in different storages. 

When we call `delete retain N` command, we expect to retain N unique backups without duplicates, so now we count them using a map.